### PR TITLE
[data grid] Use `styled` from system

### PR DESCRIPTION
**Pull request by siriwatknp**
### 1. A simple conversion from `@mui/material/styles` to `@mui/system`.
    
These component does not use Material UI theme tokens, so I think it is safe to start using `@mui/system`.

### 2. Migrate from Material UI theme token to DataGrid CSS variables. These variables are defined by a single `GridRootStyles`.

Here is what it looks like when we have Joy UI integrated:

```js
// /material/GridRootStyles.ts
export const GridRootStyles = styled('div', {
  name: 'MuiDataGrid',
  slot: 'Root',
})(({ theme }) => {
  return {
    // we could use --unstable_DataGrid-radius as a prefix at the beginning and keep refining until we are ready to ship CSS variables
    '--DataGrid-radius': theme.shape.borderRadius,
    '--DataGrid-headWeight': theme.typography.fontWeightMedium,
    '--DataGrid-overlayBackground': theme.vars
      ? `rgba(${theme.vars.palette.background.defaultChannel} / ${theme.vars.palette.action.disabledOpacity})`
      : alpha(theme.palette.background.default, theme.palette.action.disabledOpacity),
      …
  }
})


// /joy/GridRootStyles.ts
export const GridRootStyles = styled('div', {
  name: 'MuiDataGrid',
  slot: 'Root',
})(({ theme }) => {
  return {
    // we could use --unstable_DataGrid-radius as a prefix at the beginning and keep refining until we are ready to ship CSS variables
    '--DataGrid-radius': theme.vars.radius.sm,
    '--DataGrid-headWeight': theme.vars.fontWeight.md,
    '--DataGrid-overlayBackground': `rgba(${theme.vars.palette.neutral.darkChannel} / 0.5)`,
      …
  }
})
```

### CODE REVIEW

File: packages/grid/x-data-grid/src/components/GridRowCount.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import { unstable_composeClasses as composeClasses } from '@mui/utils';
-import { styled, SxProps, Theme } from '@mui/material/styles';
+import { styled, SxProps, Theme } from '@mui/system';
```
Comment (flaviendelangle) [inline packages/grid/x-data-grid/src/components/GridRowCount.tsx:5]: Based on https://github.com/mui/material-ui/issues/35840, could we use nested imports here ?

Comment (siriwatknp) [reply packages/grid/x-data-grid/src/components/GridRowCount.tsx:5]: `styled` is exported directly from the index. It does not have a nested folder after the build. We could improve it on the core but it should not block this PR.

Both of these are the same (in terms of efficiency).
```js
import { styled } from '@mui/material/styles';
import { styled } from '@mui/system';
```

**Review by cherniavskii — APPROVED** [review]
🎉

**Comment by siriwatknp** [comment]
@cherniavskii I decided to add `unstable_` prefix to the new CSS variables to implicitly keep them private at the time being. Once we have Joy UI and are ready to share this with the community, we can remove them.

Note, I think we can just use a simple string for referring to a CSS variable since the syntax `--unstable_DataGrid-*` is pretty unique.

**Pull request merged**